script.js

```diff
@@ -1,27 +1,47 @@
 const BurgerEl = document.getElementById("burger");
-const NavItemsEl = document.getElementById("navItems")
+const NavItemsEl = document.getElementById("navItems");
+const menuOpen = document.createElement("div");
+menuOpen.classList.add("menuOpen");
+const overlay = document.querySelector('#overlay');
+overlay.appendChild(menuOpen);
 
-function burgerToggle(){
-    if(NavItemsEl.style.visibility == "hidden"){
+function burgerToggle() {
+    if (NavItemsEl.style.visibility == "hidden") {
         NavItemsEl.style.visibility = "visible";
     }
-    else{
+    else {
         NavItemsEl.style.visibility = "hidden";
-    } 
+    }
 }
 
-BurgerEl.addEventListener("click", ()=>{    
-    NavItemsEl.classList.toggle("active")   
+BurgerEl.addEventListener("click", () => {
+    console.log('click open');
+    // NavItemsEl.classList.toggle("active")
+    overlay.style.display = "block";
+    menuOpen.innerHTML = `<span id='closeBtn'>&#10005;</span>
+    <ul id="menuNavItems">
+            <li class="menuNavItem">Shop</li>
+            <li class="menuNavItem">Recycle denim</li>
+            <li class="menuNavItem">Contact us</li>
+            <li class="menuNavItem">FAQ</li>
+        </ul>`
+    document.querySelector('#closeBtn').addEventListener('click', () => { closeMenu() })
```

Comment:
Is the close button created every time? Otherwise, store this as a variable.

If the event listeners are never removed, each time I click it will create a new event listener. So, theoretically, if someone opens the menu 10 times, there will be 10 event listeners.

If the DOM element is removed between toggles, there will be event listeners "floating around" in the browser memory, not connected to anything. Because we're not sure when the garbage collection runs, this might drain performance (as it has to keep track of the event listeners, and their *potential* connection to a DOM element).

So as a general rule of thumb, event listeners should only be created once, and never inside another click listener, unless there's a very specific reason for it :)